examples/inverse/plot_dipole_comparison.py

```diff
@@ -0,0 +1,136 @@
+
+This creates a small grid of source locations and performs dipole
+fits using both MNE-C and mne-python.
+"""
+# Author: Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from os import path as op
+import matplotlib.pyplot as plt
+import tempfile
+import shutil
+
+import mne
+
+print(__doc__)
+
+meg = True
+eeg = True
+
+data_path = mne.datasets.testing.data_path(download=False)
+subjects_dir = op.join(data_path, 'subjects')
+fname_evo = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-ave.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-cov.fif')
+fname_bem = op.join(subjects_dir, 'sample', 'bem',
+                    'sample-1280-1280-1280-bem-sol.fif')
+fname_dip = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_set1.dip')
+fname_mri = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-trans.fif')
+
+# Store temporary files somewhere
+temp_dir = tempfile.mkdtemp()
+fname_dip = op.join(temp_dir, 'test.dip')
+fname_src = op.join(temp_dir, 'test-src.fif')
+fname_sim = op.join(temp_dir, 'test-ave.fif')
+
+#
+# Simulate data on a 4 cm grid @ SNR=20 with random orientations
+#
+amp = 10e-9
+rng = np.random.RandomState(0)
+src = mne.setup_volume_source_space('sample', fname_src, pos=40,
+                                    bem=fname_bem, overwrite=True)
+nn = rng.randn(src[0]['np'], 3)
+nn /= np.sqrt(np.sum(nn * nn, axis=1)[:, np.newaxis])
+src[0]['nn'] = nn
+evoked = mne.read_evokeds(fname_evo)[0]
+fwd = mne.make_forward_solution(evoked.info, fname_mri, fname_src,
+                                fname_bem, n_jobs=2)
+fwd = mne.convert_forward_solution(fwd, force_fixed=True)
+cov = mne.read_cov(fname_cov)
+stc = mne.VolSourceEstimate(amp * np.eye(fwd['src'][0]['nuse']),
+                            fwd['src'][0]['vertno'], 0, 0.001)
+evoked = mne.simulation.generate_evoked(fwd, stc, evoked, cov, snr=20,
+                                        random_state=rng)
+if eeg:
+    evoked.add_proj(mne.proj.make_eeg_average_ref_proj(evoked.info))
+mne.write_evokeds(fname_sim, evoked)
+picks = mne.pick_types(evoked.info, meg=meg, eeg=eeg)
+evoked.pick_channels([evoked.ch_names[k] for k in picks])
+
+#
+# Run MNE-C version
+#
+mne.utils.run_subprocess([
+    'mne_dipole_fit', '--meas', fname_sim, '--meg', '--eeg',
+    '--bem', fname_bem, '--noise', fname_cov, '--dip', fname_dip,
+    '--mri', fname_mri, '--reg', '0'] +
+    (['--meg'] if meg else []) + (['--eeg'] if eeg else []))
+dip_c = mne.read_dipole(fname_dip)
+shutil.rmtree(temp_dir)
+
+#
+# Run mne-python version
+#
+dip = mne.fit_dipole(evoked, fname_cov, fname_bem, fname_mri, n_jobs=2)
+
+#
+# Compare to original points
+#
+trans = mne.read_trans(fname_mri)
+mne.transform_surface_to(src[0], 'head', trans)
+src_rr = src[0]['rr'][src[0]['vertno']]
+src_nn = fwd['source_nn']
+
+# MNE-C skips the last "time" point :(
+dip.crop(dip_c.times[0], dip_c.times[-1])
```

Comment:
IIRC my motivation behind doing this was first because the dipole fits for the real data were not very good (low GOF values). Then I decided I might as well make it a comparison. Is it worth showing on the sample dataset even if the GOF values are not good?